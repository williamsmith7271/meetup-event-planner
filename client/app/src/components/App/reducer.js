import update from 'react-addons-update';
import * as types from './constants';

export const initialState = {
  authToken: null,
};

const appReducer = (state = initialState, action) => {
  switch (action.type) {
    case types.SET_AUTH_TOKEN:
      return update(state, {
        authToken: {
          $set: action.authToken,
        },
      });
    default:
      return state;
  }
};

export default appReducer;
