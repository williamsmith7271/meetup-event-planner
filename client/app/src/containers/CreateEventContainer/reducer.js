import * as types from './constants';
import update from 'react-addons-update';

export const initialState = {
  error: null,
  message: null,
  guestList: [],
  geoLocation: null,
};

const createEventReducer =
  (state = initialState, action) => {
    switch (action.type) {
      case types.CREATE_EVENT_ERROR:
        return update(state, {
          error: {
            $set: action.error,
          },
        });
      case types.CREATE_EVENT_MESSAGE:
        return update(state, {
          message: {
            $set: action.message,
          },
        });
      case types.CLEAR_CREATE_EVENT_ERROR:
        return update(state, {
          error: {
            $set: null,
          },
        });
      case types.CLEAR_CREATE_EVENT_MESSAGE:
        return update(state, {
          message: {
            $set: null,
          },
        });
      case types.CREATE_EVENT_ADD_GUEST:
        return update(state, {
          guestList: {
            $set: [...state.guestList, action.guest],
          },
        });
      case types.CREATE_EVENT_REMOVE_GUEST:
        return update(state, {
          guestList: {
            $set: state.guestList.filter((_, i) => i !== action.index),
          },
        });
      default:
        return state;
    }
  };

export default createEventReducer;
