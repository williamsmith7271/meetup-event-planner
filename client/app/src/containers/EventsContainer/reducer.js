import * as types from './constants';
import update from 'react-addons-update';

export const initialState = {
  current: 10,
};

const eventsReducer =
  (state = initialState, action) => {
    switch (action.type) {
      case types.INCREMENT_CURRENT:
        return update(state, {
          current: {
            $set: state.current + 5,
          },
        });
      default:
        return state;
    }
  };

export default eventsReducer;
