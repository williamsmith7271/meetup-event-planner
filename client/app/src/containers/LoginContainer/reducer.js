import * as types from './constants';
import update from 'react-addons-update';

export const initialState = {
  error: null,
  message: null,
  isLoading: false,
};

const loginReducer =
  (state = initialState, action) => {
    switch (action.type) {
      case types.LOGIN_SHOW_ERROR:
        return update(state, {
          error: {
            $set: action.error,
          },
          isLoading: {
            $set: false,
          },
        });
      case types.LOGIN_SHOW_MESSAGE:
        return update(state, {
          message: {
            $set: action.message,
          },
          isLoading: {
            $set: false,
          },
        });
      case types.LOGIN_CLEAR_ERROR:
        return update(state, {
          error: {
            $set: null,
          },
        });
      case types.LOGIN_CLEAR_MESSAGE:
        return update(state, {
          message: {
            $set: null,
          },
        });
      case types.LOGIN_SET_LOADING:
        return update(state, {
          isLoading: {
            $set: true,
          },
        });
      default:
        return state;
    }
  };

export default loginReducer;
