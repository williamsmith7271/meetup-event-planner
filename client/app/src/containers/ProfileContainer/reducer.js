import * as types from './constants';
import update from 'react-addons-update';

export const initialState = {
  isEditing: false,
  bioInput: null,
  error: null,
  isLoading: false,
  avatarInput: null,
  emailInput: null,
  employerInput: null,
};

const profileReducer =
  (state = initialState, action) => {
    switch (action.type) {
      case types.PROFILE_EDIT_AVATAR:
        return update(state, {
          avatarInput: {
            $set: action.avatar,
          },
        });
      case types.PROFILE_EDIT_BIO:
        return update(state, {
          bioInput: {
            $set: action.bio,
          },
        });
      case types.PROFILE_EDIT_EMAIL:
        return update(state, {
          emailInput: {
            $set: action.email,
          },
        });
      case types.PROFILE_EDIT_EMPLOYER:
        return update(state, {
          employerInput: {
            $set: action.employer,
          },
        });
      case types.PROFILE_SUBMISSION_INITIATION:
        return update(state, {
          isLoading: {
            $set: true,
          },
        });
      case types.PROFILE_SUBMISSION_SUCCESS:
        return update(state, {
          isEditing: {
            $set: false,
          },
          bioInput: {
            $set: null,
          },
          employerInput: {
            $set: null,
          },
          emailInput: {
            $set: null,
          },
          avatarInput: {
            $set: null,
          },
          isLoading: {
            $set: false,
          },
        });
      case types.PROFILE_SUBMISSION_FAILURE:
        return update(state, {
          isEditing: {
            $set: false,
          },
          bioInput: {
            $set: null,
          },
          avatarInput: {
            $set: null,
          },
          employerInput: {
            $set: null,
          },
          emailInput: {
            $set: null,
          },
          isLoading: {
            $set: false,
          },
          error: {
            $set: action.error,
          },
        });
      case types.PROFILE_CLEAR_ERROR:
        return update(state, {
          error: {
            $set: null,
          },
        });
      case types.PROFILE_CANCEL_EDITING:
        return update(state, {
          isEditing: {
            $set: false,
          },
          bioInput: {
            $set: null,
          },
          avatarInput: {
            $set: null,
          },
          employerInput: {
            $set: null,
          },
          emailInput: {
            $set: null,
          },
        });
      case types.PROFILE_START_EDITING:
        return update(state, {
          isEditing: {
            $set: true,
          },
        });
      case types.PROFILE_SET_DEFAULT_INPUTS:
        return update(state, {
          bioInput: {
            $set: action.inputs.bio,
          },
          avatarInput: {
            $set: action.inputs.avatar,
          },
          employerInput: {
            $set: action.inputs.employer,
          },
          emailInput: {
            $set: action.inputs.email,
          },
        });
      default:
        return state;
    }
  };

export default profileReducer;
