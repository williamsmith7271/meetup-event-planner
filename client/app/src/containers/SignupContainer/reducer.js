import * as types from './constants';
import update from 'react-addons-update';

export const initialState = {
  error: null,
  message: null,
};

const signupReducer =
  (state = initialState, action) => {
    switch (action.type) {
      case types.SIGNUP_SHOW_ERROR:
        return update(state, {
          error: {
            $set: action.error,
          },
        });
      case types.SIGNUP_SHOW_MESSAGE:
        return update(state, {
          message: {
            $set: action.message,
          },
        });
      case types.SIGNUP_CLEAR_ERROR:
        return update(state, {
          error: {
            $set: null,
          },
        });
      case types.SIGNUP_CLEAR_MESSAGE:
        return update(state, {
          message: {
            $set: null,
          },
        });
      default:
        return state;
    }
  };

export default signupReducer;
