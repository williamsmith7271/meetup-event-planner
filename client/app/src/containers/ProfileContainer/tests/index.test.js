import Profile from '../index';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import React from 'react';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { initialState as profileContainer } from '../reducer';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

describe('<Profile />', () => {
  it('should render with default props', () => {
    const store = mockStore({ profileContainer });
    const wrapper = shallow(
      <Profile store={store} />
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
