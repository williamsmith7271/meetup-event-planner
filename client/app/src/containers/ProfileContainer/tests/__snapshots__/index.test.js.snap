exports[`<Profile /> should render with default props 1`] = `
<Apollo(Apollo(Profile))
  actions={
    Object {
      "authenticateUser": [Function anonymous],
      "invalidateUser": [Function anonymous],
      "loadPersistedUser": [Function anonymous],
      "logoutUser": [Function anonymous],
      "profileCancelEditing": [Function anonymous],
      "profileClearError": [Function anonymous],
      "profileEditAvatar": [Function anonymous],
      "profileEditBio": [Function anonymous],
      "profileEditEmail": [Function anonymous],
      "profileEditEmployer": [Function anonymous],
      "profileStartEditing": [Function anonymous],
      "profileSubmissionFailure": [Function anonymous],
      "profileSubmissionInitiation": [Function anonymous],
      "profileSubmissionSuccess": [Function anonymous],
      "setDefaultInputs": [Function anonymous],
      "setPersistentUser": [Function anonymous]
    }
  }
  avatarInput={null}
  bioInput={null}
  emailInput={null}
  employerInput={null}
  isEditing={false}
  isLoading={false}
  store={
    Object {
      "clearActions": [Function clearActions],
      "dispatch": [Function anonymous],
      "getActions": [Function getActions],
      "getState": [Function getState],
      "replaceReducer": [Function replaceReducer],
      "subscribe": [Function subscribe]
    }
  }
  submissionError={null}
  user={
    Object {
      "authToken": null
    }
  } />
`;
